### PR TITLE
add client.RequestToJoinTeam

### PR DESCRIPTION
**Pull request by idrysdale**

### CODE REVIEW

**Comment by paulfurley** [comment]
Rebasing

File: api/api.go [hunk 1]
```diff
@@ -230,6 +230,28 @@ func (c *Client) UpsertPublicKey(armoredPublicKey string, privateKey *pgpkey.Pgp
 	return err
 }
 
+// CreateRequestToJoinTeam posts a request to join the team identified by the UUID with the
+// given fingerprint and email
+func (c *Client) CreateRequestToJoinTeam(
```
Comment (paulfurley) [inline api/api.go]: or should it be `RequestToJoinTeam` ?

File: api/api.go [hunk 2]
```diff
@@ -248,6 +248,28 @@ func (c *Client) GetTeamName(teamUUID uuid.UUID) (string, error) {
...
+	path := fmt.Sprintf("team/%s/requests-to-join", teamUUID)
+	requestToJoinTeamRequest := v1structs.RequestToJoinTeamRequest{TeamEmail: email}
+
+	request, err := c.newRequest("POST", path, requestToJoinTeamRequest)
+	if err != nil {
+		return err
+	}
+	request.Header.Add("authorization", authorization(fpr))
+
+	_, err = c.do(request, nil)
```
Comment (paulfurley) [inline api/api.go]: Currently no handling of the special `409` cases, is that OK?

File: api/api_test.go [hunk 3]
```diff
@@ -396,6 +397,65 @@ func TestGetTeamName(t *testing.T) {
...
+		defer teardown()
+
+		mockResponseHandler := func(w http.ResponseWriter, r *http.Request) {
+			assertClientSentVerb(t, "POST", r.Method)
+			gotRequest := new(v1structs.RequestToJoinTeamRequest)
+			json.NewDecoder(r.Body).Decode(gotRequest)
+			assert.Equal(t, expectedRequest, gotRequest)
+
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprint(w, `{"detail": "can't write to database"}`)
```
Comment (paulfurley) [inline api/api_test.go]: nice!

**Comment by paulfurley** [comment]
i broke the tests in the rebase, fixed now

File: api/api.go (hunk 1 above)
Comment (idrysdale) [reply api/api.go]: Yeah - that's better.

File: api/api.go (hunk 2 above)
Comment (idrysdale) [reply api/api.go]: I've added that now.

**Comment by paulfurley** [comment]
rebased

**Pull request merged**